system/jlib/jutil.cpp

```diff
@@ -3349,10 +3349,10 @@ const char * generatePassword(StringBuffer &pwd, int pwdLen)
     const char alphaUC[] = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
     const char alphaLC[] = "abcdefghijklmnopqrstuvwxyz";
     const char numeric[] = "0123456789";
-    const char symbol[] = "~@#$%^*()_-+={[}]:,.?";
+    const char symbol[] = "~@#%^*_-+{[}]:,.?";
 
     //Ensures each character group used at least once
-    srand(time(0));
+    srand(unsigned(GetCurrentProcessId() + GetCurrentThreadId() + time(0)));
```

Comment:
@RussWhitehead You could use urandom which uses system device entropy to generate random numbers.

https://linux.die.net/man/4/urandom

We use it for the global transaction ids here https://github.com/hpcc-systems/HPCC-Platform/blob/870834b08a0550db458f06d0ef3dfe062c018e2c/system/globalid/lnuid.cpp#L91

But there are probably a few ways to call it.

Reply:
@afishbeck Wouldn't spawning a process to do this be extremely slow and somewhat  overkill?
Will
GetCurrentProcessId() + GetCurrentThreadId() + time(0)
will be both fast and achieve the randomness needed?

Reply:
@RussWhitehead Not sure what you mean about spawning a process.  Our createUniqueId() function doesn't spawn anything.
/dev/random (better for strong cryptography) can block which is why we use /dev/urandom which doesn't.

Reply:
My bad, I thought it was spawned. Still, it's file IO which seems very expensive?